orchagent/main.cpp

```diff
@@ -453,9 +455,17 @@ int main(int argc, char **argv)
     DBConnector config_db("CONFIG_DB", 0);
     DBConnector state_db("STATE_DB", 0);
 
-    // Instantiate ZMQ server 
-    SWSS_LOG_NOTICE("Instantiate ZMQ server : %s", zmq_server_address.c_str());
-    ZmqServer zmq_server(zmq_server_address.c_str());
+    // Instantiate ZMQ server
+    shared_ptr<ZmqServer> zmq_server = nullptr;
+    if (enable_zmq)
+    {
+        SWSS_LOG_NOTICE("Instantiate ZMQ server : %s", zmq_server_address.c_str());
+        zmq_server = make_shared<ZmqServer>(zmq_server_address.c_str());
```

Comment:
>zmq_server_address [](http://example.com/codeflow?start=44&length=18)

If `zmq_server_address` is an invalid argument, the constructor will not throw, but later "ZmqServer::mqPollThread" will throw. Will orchagent catch this exception and print error messages?

Reply:
In the "ZmqServer::mqPollThread" it's already log error to syslog before throw.
Because it's in the thread that exception will terminate the orchagent.

If it's necessary, I can add a global exception handler to catch and handle exception.